vignettes/IRangesOverview.Rmd

```diff
@@ -0,0 +1,527 @@
+
+## Aggregating Views
+
+While `sapply` can be used to loop over each window, the native functions
+`viewMaxs`, `viewMins`, `viewSums`, and `viewMeans` provide fast looping to
+calculate their respective statistical summaries.
+
+```{r views-looping}
+head(viewSums(xViews))
+viewSums(xViewsList)
+head(viewMaxs(xViews))
+viewMaxs(xViewsList)
+```
+
+# Lists of Atomic Vectors
+
+In addition to the range-based objects described in the previous sections, the
+`r Biocpkg("IRanges")` package provides containers for storing lists of atomic
+vectors such as *integers* or *Rle* objects. The *IntegerList* and *RleList*
+classes represent lists of *integer* vectors and *Rle* objects respectively.
+They are subclasses of the *AtomicList* virtual class which is itself a subclass
+of the *List* virtual class defined in the `r Biocpkg("S4Vectors")` package.
+
+```{r AtomicList-intro}
+showClass("RleList")
+```
+
+As the class definition above shows, the *RleList* class is virtual with
+subclasses *SimpleRleList* and *CompressedRleList*. A *SimpleRleList* class uses
+an ordinary *R* list to store the underlying elements and *CompressedRleList*
+the class stores the elements in an unlisted form and keeps track of where the
+element breaks are. The former "simple list" class is useful when the *Rle*
+elements are long and the latter "compressed list" class is useful when the list
+is long and/or sparse (i.e. a number of the list elements have length 0).
+
+In fact, all of the atomic vector types (logical, integer, numeric, complex,
+character, raw, and factor) have similar list classes that derive from the
+*List* virtual class. For example, there is an *IntegerList* virtual class with
+subclasses *SimpleIntegerList* and *CompressedIntegerList*.
+
+Each of the list classes for atomic sequences, be they stored as vectors or
+*Rle* objects, have a constructor function with a name of the appropriate list
+virtual class, such as *IntegerList*, and an optional argument `compress` that
+takes an argument to specify whether or not to create the simple list object
+type or the compressed list object type. The default is to create the compressed
+list object type.
+
+```{r list-construct}
+args(IntegerList)
+cIntList1 <- IntegerList(x=xVector, y=yVector)
+cIntList1
+sIntList2 <- IntegerList(x=xVector, y=yVector, compress=FALSE)
+sIntList2
+## sparse integer list
+xExploded <- lapply(xVector[1:5000], function(x) seq_len(x))
+cIntList2 <- IntegerList(xExploded)
+sIntList2 <- IntegerList(xExploded, compress=FALSE)
+object.size(cIntList2)
+object.size(sIntList2)
+```
+
+The `length` function returns the number of elements in a *Vector*-derived
+object and, for a *List*-derived object like "simple list" or "compressed list",
+the `lengths` function returns an integer vector containing the lengths of each
+of the elements:
+
+```{r list-length}
+length(cIntList2)
+Rle(lengths(cIntList2))
+```
+
+Just as with ordinary *R* objects, *List*-derived object support the `[[` for
```

Comment:
Missing a word probably due `\Rclass{list}` not being rendered correctly. Change to

    Just as with ordinary *R*  *List* objects, *List*-derived object support the `[[` for